### PR TITLE
#1 basic structure

### PR DESCRIPTION
**Pull request by Digvorzhak**
All packages configured

### CODE REVIEW

File: client/package.json [hunk 1]
```diff
@@ -21,6 +25,7 @@
     "eslint-plugin-react": "^7.32.2",
     "eslint-plugin-react-hooks": "^4.6.0",
     "eslint-plugin-react-refresh": "^0.3.4",
+    "styled-components": "^5.3.6",
```
Comment (yishain11) [inline client/package.json:28]: styled components is not dev dependancy, its a regular dep

File: client/src/ErrorBoundary.jsx [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+class ErrorBoundary extends React.Component {
+  state = {hasError: false}
+
+  static getDerivedStateFromError(error){
+    return {hasError: true}
+  }
+
+  componentDidCatch(error, info){
+    console.log(error, info)
```
Comment (yishain11) [inline client/src/ErrorBoundary.jsx:11]: console.error, not console.log

File: client/src/ErrorBoundary.jsx [hunk 3]
```diff
@@ -0,0 +1,22 @@
+import React from "react";
+
+class ErrorBoundary extends React.Component {
```
Comment (yishain11) [inline client/src/ErrorBoundary.jsx:3]: we are not working with class components, change to functional, and use the react error bounderies library from npm

**Review by yishain11 — CHANGES_REQUESTED** [review]

File: client/src/ErrorBoundary.jsx [hunk 4]
```diff
@@ -0,0 +1,17 @@
+import React, { useState } from "react";
```
Comment (yishain11) [inline client/src/ErrorBoundary.jsx:1]: remove import react

File: client/src/main.jsx [hunk 5]
```diff
@@ -1,10 +1,16 @@
 import React from "react";
 import ReactDOM from "react-dom/client";
 import App from "./App.jsx";
-import "./index.css";
+import {store} from './app/store.jsx'
+import { Provider } from "react-redux";
+import ErrorBoundary from "./ErrorBoundary.jsx";
```
Comment (yishain11) [inline client/src/main.jsx:6]: you are not using error boundery package. please import it and use it - and inside it use the component you created. see docs - https://github.com/bvaughn/react-error-boundary

**Review by yishain11 — CHANGES_REQUESTED** [review]

File: client/index.html [hunk 6]
```diff
@@ -2,9 +2,9 @@
 <html lang="en">
   <head>
     <meta charset="UTF-8" />
-    <link rel="icon" type="image/svg+xml" href="/vite.svg" />
+    <!-- <link rel="icon" type="image/svg+xml" href="/vite.svg" /> -->
```
Comment (yishain11) [inline client/index.html:5]: remove commented out code

File: client/src/main.jsx [hunk 7]
```diff
@@ -1,10 +1,18 @@
...
 import React from "react";
 import ReactDOM from "react-dom/client";
 import App from "./App.jsx";
-import "./index.css";
+import {store} from './app/store.jsx'
+import { Provider } from "react-redux";
 
 ReactDOM.createRoot(document.getElementById("root")).render(
   <React.StrictMode>
+    <ErrorBoundary fallback={<div>Something went wrong</div>}>
```
Comment (yishain11) [inline client/src/main.jsx:12]: instead of the div, create external error component and import it. this will make it easier to edit it in the future

**Review by yishain11 — CHANGES_REQUESTED** [review]

File: client/src/main.jsx (hunk 7 above)
Comment (Digvorzhak) [reply client/src/main.jsx]: done

File: client/src/components/styles/GlobalStyles.jsx [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+
+@media only screen and (max-width: 800px) {
+  html {
+    font-size: 50%;
+  }
+}
+
+body {
+  box-sizing: border-box;
+  font-family: 'Open Sans', sans-serif;
```
Comment (yishain11) [inline client/src/components/styles/GlobalStyles.jsx:27]: use the font familywe are accualty using -ABEEZ or somwthing like it

**Review by yishain11 — CHANGES_REQUESTED** [review]

**Pull request merged**